elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -118,59 +132,108 @@ public void readAndValidateConfigs() throws IOException {
+                        .filter(entry -> entry.getKey().startsWith(Config.SECURITY.getConfigPath()))
+                        .map(entry -> {
+                            try {
+                                String content = IOUtils.toString(entry.getValue().getInputStream(), UTF_8);
+                                validateConfigForMissingVariables(content, this.modelVariables);
+                                return DynamicConfigHelpers.stringToElideSecurityPojo(content, this.modelVariables);
+                            } catch (IOException e) {
+                                throw new IllegalStateException(e);
+                            }
+                        })
+                        .findAny()
+                        .orElse(new ElideSecurityConfig());
     }
 
     /**
-     * Read table config files and checks for any missing Handlebar variables.
-     * @throws IOException
+     * Read and validates db config files.
+     * @return Set<DBConfig> Set of SQL DB Configs
      */
-    private void readTableConfig() throws IOException {
-        Set<Table> tables = new HashSet<>();
-        if (this.resourceMap.isEmpty()) {
-            throw new IllegalStateException("No Table configs found at: " + this.configDir);
-        }
-        for (Entry<String, Resource> entry : this.resourceMap.entrySet()) {
-            String content = IOUtils.toString(entry.getValue().getInputStream(), StandardCharsets.UTF_8);
-            validateConfigForMissingVariables(content, this.variables);
-            ElideTableConfig table = DynamicConfigHelpers.stringToElideTablePojo(content, this.variables);
-            tables.addAll(table.getTables());
+    private Set<DBConfig> readDbConfig(Config config) {
+
+        return this.resourceMap
+                        .entrySet()
+                        .stream()
+                        .filter(entry -> entry.getKey().startsWith(config.getConfigPath())).map(entry -> {
+                            try {
+                                String content = IOUtils.toString(entry.getValue().getInputStream(), UTF_8);
+                                validateConfigForMissingVariables(content, this.dbVariables);
+                                return DynamicConfigHelpers.stringToElideDBConfigPojo(content, this.dbVariables);
+                            } catch (IOException e) {
+                                throw new IllegalStateException(e);
+                            }
+                        })
+                        .flatMap(dbconfig -> dbconfig.getDbconfigs().stream())
+                        .collect(Collectors.toSet());
+    }
+
+    /**
+     * Read and validates table config files.
+     */
+    private Set<Table> readTableConfig() {
+
+        Set<Table> tables = this.resourceMap
+                        .entrySet()
+                        .stream()
+                        .filter(entry -> entry.getKey().startsWith(Config.TABLE.getConfigPath())).map(entry -> {
```

Comment:
map should go on its own line separate from filter.

Reply:
updated